oeps/oep-0052-arch-event-bus-technology.rst

```diff
@@ -0,0 +1,264 @@
+
+or
+
+* `Confluent Platform`_ - Enterprise Kafka Distribution (Open Source, Community, or Commercial)
+
+  * `Schema Registry <https://www.confluent.io/product/confluent-platform/data-compatibility/>`__
+  * Monitoring and alerting capabilities (Commercial)
+  * Self-balancing clusters (Commercial)
+  * Tiered storage (Commercial) (future feature of Apache Kafka)
+  * Infinite retention (Cloud only?)
+
+Additional Notes:
+
+* `Apache Pulsar`_ has similar features as part of its platform, which is why it makes a good potential alternative. However, the features are less battle-tested and the deployment story *may* be more complicated.
+* Confluent also offers Confluent Cloud, a fully managed solution that offers much simpler operations, but is unlikely to be used by edX.org.
+
+Also see a useful and biased `comparison of Apache Kafka vs Vendors`_ by Kai Waehner (of Confluent), comparing various providers and distributions of Kafka and related or competitive services.
+
+.. _Amazon MSK: https://aws.amazon.com/msk/
+.. _Confluent Platform: https://www.confluent.io/product/confluent-platform
+.. _comparison of Apache Kafka vs Vendors: https://www.kai-waehner.de/blog/2021/04/20/comparison-open-source-apache-kafka-vs-confluent-cloudera-red-hat-amazon-msk-cloud/
+
+Kafka Highlights
+~~~~~~~~~~~~~~~~
+
+Pros
+^^^^
+
+* Battle-tested, widely adopted, big community, lots of documentation and answers.
+* Amazon MSK (AWS service) provides hosted path of least resistance.
+* `New Relic integration with Amazon MSK`_ (useful to edX.org).
+
+Cons
+^^^^
+
+* Many open questions about add-ons required for developers and operators.
+* Complex to manage, including likely manual scaling.
+
+.. _New Relic integration with Amazon MSK: https://docs.newrelic.com/docs/integrations/amazon-integrations/aws-integrations-list/aws-managed-kafka-msk-integration/
+
+Consequences
+------------
+
+* Operators will need to deploy and manage the selected infrastructure, which is likely to be complex. If Apache Kafka is selected, there are likely to be a set of auxiliary parts to provide all required functionality for our message bus.
+* Education will be required for both developers and operators regarding best practices for each role.
+* Code to interact with Kafka and its libraries will be added to core services.
+* At least one initial use case must be completed. One potential candidate is the grade change event in the LMS, and its use by the Credentials service.
+* Once we have a message bus, we can investigate other potential use cases:
+
+  * Course/program update propagation.
+  * Feed into xAPI/Caliper capabilities.
+  * New services and features can be built fully de-coupled from the core application.
+
+Rejected Alternatives
+---------------------
+
+Apache Pulsar
+~~~~~~~~~~~~~
+
+Although rejected to start, `Apache Pulsar`_ remains an option if solving with Kafka turns out to be overly burdensome for developers or operators.
```

Comment:
@blarghmatey: Thanks for your comment: https://github.com/edx/open-edx-proposals/pull/233#issuecomment-913605682. I’m going to move discussion of Pulsar to this section, so we can have a threaded discussion.
1. I think some of what you had written about is already discussed here, but I can adjust accordingly.
2. The one detail I almost missed from your post was about multi-tenancy.  I appreciate you bringing it up.  This may require more research. Typically, there is a way to solve most issues with Kafka, but I’m not sure how ugly or elegant this would be.